File: backend/app/schemas/data_dict.py
from pydantic import BaseModel, Field


class DataDictCreate(BaseModel):
    dict_type: str
    code: str
    label: str
    value: dict = Field(default_factory=dict)
    sort_order: int = 0
    enabled: bool = True
    description: str | None = None


class DataDictUpdate(BaseModel):
    label: str | None = None
    value: dict | None = None
    sort_order: int | None = None
    enabled: bool | None = None
    description: str | None = None


class DataDictResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: str
    dict_type: str
    code: str
    label: str
    value: dict = Field(default_factory=dict)
    scope: str
    enterprise_id: str | None = None
    sort_order: int = 0
    enabled: bool = True
    is_system: bool = False
    description: str | None = None

File: frontend/src/types/dataDict.ts
/** 数据字典类型定义（与 backend/app/schemas/data_dict.py 对应）。 */

export interface DataDictItem {
  id: string;
  dict_type: string;
  code: string;
  label: string;
  value: Record<string, unknown>;
  scope: "system" | "enterprise";
  enterprise_id: string | null;
  sort_order: number;
  enabled: boolean;
  is_system: boolean;
  description?: string | null;
}

export interface DataDictPayload {
  dict_type: string;
  code: string;
  label: string;
  value: Record<string, unknown>;
  sort_order?: number;
  enabled?: boolean;
  description?: string | null;
}
